### PR TITLE
[Filesharing] - Expire Upload Errors after 10 seconds OR when more file are uploaded.

### PR DESCRIPTION
**Pull request by anjulgarg**
# What
- Add new internal component `SendBoxErrorBar` for displaying SendBox errors with timetout
- Remove a file upload error message above sendbox after 10 seconds.
- Adding more files to the sendbox will remove existing erroneous file uploads. And thus removing an error message being displayed above sendbox
- Adds tests for AzureCommunicationFileUploadAdapter

# Why
Part of filesharing error handling

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/SendBoxErrorBar.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * @internal
+ */
+export interface SendBoxErrorBarProps {
+  /** Error message to render */
+  message?: string;
+  /**
+   * Timeout in ms after which the error bar disappears.
+   * If undefined, the error bar will never disappear.
+   */
+  timeout?: number;
```
Comment (JamesBurnside) [inline packages/react-components/src/components/SendBoxErrorBar.tsx:17]: Why do we remove the error message after a timeout? Do we do this to any other error messages?

File: packages/react-components/src/components/SendBoxErrorBar.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  useEffect(() => {
+    if (timeout !== undefined) {
+      const messageTimeout = setTimeout(() => {
+        setErrorMessage(undefined);
+      }, timeout);
+      return () => {
+        clearTimeout(messageTimeout);
+      };
+    } else {
+      return;
```
Comment (JamesBurnside) [inline packages/react-components/src/components/SendBoxErrorBar.tsx:36]: nit: remove empty else

**Review by JamesBurnside — APPROVED** [review]

File: packages/react-components/src/components/SendBoxErrorBar.tsx (hunk 1 above)
Comment (anjulgarg) [reply packages/react-components/src/components/SendBoxErrorBar.tsx:17]: The reason is a mix of product and design requirements and mimics the file upload error behavior in teams.

File: packages/react-components/src/components/SendBoxErrorBar.tsx (hunk 2 above)
Comment (anjulgarg) [reply packages/react-components/src/components/SendBoxErrorBar.tsx:36]: Added it to remove the warning, `not all code paths return`

**Review by prabhjot-msft — APPROVED** [review]

**Pull request merged**